crates/mun_memory/src/gc/mark_sweep.rs

```diff
@@ -225,8 +226,8 @@ where
     }
 }
 
-fn alloc_obj(ty: Type) -> Pin<Box<ObjectInfo>> {
-    let ptr = NonNull::new(unsafe { std::alloc::alloc(ty.value_layout()) })
+fn alloc_struct(ty: Type) -> Pin<Box<ObjectInfo>> {
+    let ptr = NonNull::new(unsafe { std::alloc::alloc_zeroed(ty.value_layout()) })
```

Comment:
Is there a specific reason you changed this? I assume it adds a little bit of overhead that is not really needed.

Reply:
Either you need to manually ensure that you zero initialize fields that can't map their value or just zero-initialize the whole struct at the start and it turns into a noop. That seemed more efficient.